### PR TITLE
feat(main): add reserve protocol app

### PR DESCRIPTION
**Pull request by julianmrodri**
## Description

Adding the reserve-protocol app for integration with Reserve. The idea is to show the staking positions showing in one group the balance of the staked token. I another group have the staking cooldown information, which includes tokens locked and claimable.

Metadata:

* App name: `Reserve Protocol'
* Description: The Reserve Protocol allows permissionless creation of asset-backed, yield-bearing & overcollateralized stablecoins on Ethereum.
* Tags: defi stablecoins reserve
* Website (URL): https://register.app/

## Checklist

- [X] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:
- [ ] (optional) As a contributor, my Twitter handle is:

## How to test?

* Enable app: reserve-protocol
* AppToken: -> http://localhost:5001/apps/reserve-protocol/tokens?groupIds[]=staked-rsr&network=ethereum

* Cooldown Positions: -> http://localhost:5001/apps/reserve-protocol/positions?groupIds[]=cooldown&network=ethereum

* Balances Locked: -> http://localhost:5001/apps/reserve-protocol/balances?groupIds[]=cooldown&network=ethereum&addresses[]=0x172fe8a4511d7d5fa4e36a962a333c80395fd644

### CODE REVIEW

File: src/apps/reserve-protocol/ethereum/reserve-protocol.cooldown.contract-position-fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+  // async getStatsItems({ contractPosition: { dataProps } }: GetDisplayPropsParams<StakedRsr, StakedRsrDataProps>) {
+  //   return [{ label: 'In Cooldown Total Balance', value: buildNumberDisplayItem(dataProps.inCoolDownTotalBalance) }];
+  // }
+
+  async getLabel({ contractPosition }: GetDisplayPropsParams<StakedRsr>) {
+    return getLabelFromToken(contractPosition.tokens[0]);
+  }
+
+  async getTokenBalancesPerPosition({ address, contract }: GetTokenBalancesParams<StakedRsr>): Promise<BigNumberish[]> {
```
[the remark this answers is not in the document]
Comment (julianmrodri) [reply src/apps/reserve-protocol/ethereum/reserve-protocol.cooldown.contract-position-fetcher.ts:101]: Interesting why is this? Whats the purpose of Multicall?

File: src/apps/reserve-protocol/ethereum/reserve-protocol.staked-rsr.queries.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { gql } from 'graphql-request';
+
+const minValidSupply = 10 ** 18;
+
+export type RTokens = {
```
[the remark this answers is not in the document]
Comment (julianmrodri) [reply src/apps/reserve-protocol/ethereum/reserve-protocol.staked-rsr.queries.ts:5]: Will do.

File: src/apps/reserve-protocol/ethereum/reserve-protocol.cooldown.contract-position-fetcher.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (julianmrodri) [reply src/apps/reserve-protocol/ethereum/reserve-protocol.cooldown.contract-position-fetcher.ts:101]: Done.

[the remark this answers is not in the document]
Comment (wpoulin) [reply src/apps/reserve-protocol/ethereum/reserve-protocol.cooldown.contract-position-fetcher.ts:101]: Do you know what multicall is ? https://github.com/mds1/multicall

TLDR: It bundles a bunch of RPC call together in order to reduce the total amount of back and forth required to get the onchain data

**Review by wpoulin — APPROVED** [review]

**Comment by SamIam-0x** [comment]
Hello @julianmrodri  - thanks for submitting this! 

qq- do you have an ENS or address you'd want us to attribute this contribution to? We are building out a leaderboard to give credit to users who contributed to Zapper, with potential bonus points of value in the future.

Let me know, thanks!
Sam
